backend/scripts/rerun_failing_test_executions.py

```diff
@@ -0,0 +1,87 @@
+# ruff: noqa: T201
+
+import argparse
+import re
+
+import requests
+
+TO_API_URL = "https://test-observer-api.canonical.com"
+
+
+def main(artefact_id: int, test_case_regex: str, reversed: bool):
+    artefact_builds = requests.get(
+        f"{TO_API_URL}/v1/artefacts/{artefact_id}/builds"
+    ).json()
+
+    test_case_matcher = re.compile(test_case_regex)
```

Comment:
What if this call fails? Should we handle an exception here?

Reply:
By the call do you mean the api call or `re.compile` call? Failing the former would most likely a network error (not sure we should handle that). I'm not sure if we can fail the latter based on my understanding of the [docs](https://docs.python.org/3.10/library/re.html#re.compile).